tasks/omnibus.py

```diff
@@ -69,12 +69,18 @@ def bundle_install_omnibus(ctx, gem_path=None, env=None, max_try=2):
 
         with gitlab_section("Bundle install omnibus", collapsed=True):
             for trial in range(max_try):
+                # The exception being raised when ommiting `warn=True` only includes
+                # the last 10 lines of stdout/stderr which might not be enough
+                # for should_retry_bundle_install to do its job
                 res = ctx.run(cmd, env=env, warn=True, err_stream=sys.stdout)
                 if res.ok:
                     return
                 if not should_retry_bundle_install(res):
-                    return
+                    print(f'Fatal error while installing omnibus: {res.stdout}. Cannot continue.')
+                    raise Exit(code=2)
                 print(f"Retrying bundle install, attempt {trial + 1}/{max_try}")
+            print('Too many attempts at installing omnibus, giving up')
+            raise Exit(code=3)
```

Comment:
Question: Are these error codes arbitrary? It caught my eye that you decided on different codes for each exit.

Reply:
They are, my personal rule of thumb is to use a different one for each error in the absence of a clear convention, in order to easily associate an error with its source, but I can definitely unify everything to use Exit(1) since we have associated logs here

Reply:
I don't have a strong opinion since we're not really using them. If we were, I guess I'd have a set of constants / enum somewhere so that it's not just random numbers around.

Reply:
I just realized that you can simply provide an error message :sweat_smile: no more arbitrary error codes